pkg/crc/preflight/preflight_check_tray_darwin.go

```diff
@@ -100,22 +52,27 @@ func fixDaemonPlistFileExists() error {
 	if err != nil {
 		return err
 	}
-	daemonConfig := AgentConfig{
+	daemonConfig := launchd.AgentConfig{
+		Label:          daemonAgentLabel,
 		BinaryPath:     currentExecutablePath,
 		StdOutFilePath: stdOutFilePathDaemon,
+		Args:           []string{"daemon", "--log-level", "debug"},
 	}
 
 	logging.Debug("Creating daemon plist")
-	err = createPlist(daemonPlistTemplate, daemonConfig, daemonPlistFilePath)
+	err = launchd.CreatePlist(daemonConfig, daemonPlistFilePath)
```

Comment:
They can be formed from the labels, but keeping them as constants just reads better i think..

Reply:
`preflight_check_tray_darwin.go` does not need to know that plist files go to `$HOME/Library/LaunchAgents`. That's an implementation detail of the launch package. Once the path is hidden from `preflight_check_tray_darwin.go`, we can remove plist names from the public interface since they are easy to autogenerate.

Reply:
see https://github.com/cfergeau/crc/commit/db96e3c72e4a279a7a184925f44a576a78210dbc

Reply:
> see [cfergeau@db96e3c](https://github.com/cfergeau/crc/commit/db96e3c72e4a279a7a184925f44a576a78210dbc)

Which probably needs a bit more work to use the `daemonAgentLabel` `trayAgentLabel` constants rather than hardcoding these :)

Reply:
> >    see cfergeau@db96e3c

> Which probably needs a bit more work to use the daemonAgentLabel trayAgentLabel constants rather than hardcoding these :)

Looks like it needs only small change, i can cherry pick your commit and do that work on top, Thanks!